### PR TITLE
Flyout Menu - Ready 

### PR DESCRIPTION
**Pull request by NicollGuarnizo**
Demo

Flyout Menu referred in this [issue](https://github.com/eggheadio/egghead-next/issues/722) : 
   1. Meet all the functionality asked in the issue
   2. Button is hidden in mobile version (as the current Browse button)
   3. Design for light&dark mode

### CODE REVIEW

**Comment by MaggieAppleton** [comment]
This looks great!

The only thing we might want to consider adding here is tracking so we get to get data about how much people are using it.

We have a `track` component in `utils/analytics` that we can pass to each link. Here's an example of it being used on the main jumbotron: https://github.com/eggheadio/egghead-next/blob/fbe23ecce50a84ca89bb671a6fa8130fb41e8265/src/components/pages/home/jumbotron/index.tsx#L66

**Comment by NicollGuarnizo** [comment]
Track components added

File: src/components/app/header/index.tsx [hunk 1]
```diff
@@ -345,3 +346,155 @@ const IconX = () => (
...
+                        onClick={() => track(`clicked topic`, {resource: item.href})}
+                        className="flex items-start rounded-lg transition ease-in-out duration-150 hover:bg-gray-100 dark:hover:bg-gray-900 py-2 px-3"
+                      >
+                        <p className="text-base font-medium text-gray-700 transition ease-in-out duration-150 dark:text-white hover:text-black">
+                          {item.name}
+                        </p>
+                      </a>
+                    ))}
+                    <div className="mr-6 text-base font-medium  transition ease-in-out duration-150 rounded-lg hover:bg-gray-100 dark:hover:bg-gray-900 py-2 px-3">
+                      <a href="/topics" className="text-blue-500 onClick={() => track(`clicked all topics`, {location})}">
```
Comment (MaggieAppleton) [inline src/components/app/header/index.tsx:470]: Looks like the `onClick={() => track(clicked all topics, {location})}` function is declared inside `className` here

File: src/components/app/header/index.tsx [hunk 2]
```diff
@@ -345,3 +346,155 @@ const IconX = () => (
...
+    {
+      name: 'Python',
+      href: '/q/python',
+    },
+    {
+      name: 'Go',
+      href: '/q/go',
+    },
+  ]
+  const callsToAction = [
```
Comment (MaggieAppleton) [inline src/components/app/header/index.tsx:406]: `callsToAction` doesn't feel like quite the right name here. Something like `contentTypes` or `contentTypeLinks` would be more accurate. Small nitpick, but clear naming helps keep this readable.

File: src/components/app/header/index.tsx [hunk 3]
```diff
@@ -345,3 +346,155 @@ const IconX = () => (
...
+                className={classNames(
+                  open
+                    ? 'hover:bg-gray-100 dark:hover:bg-gray-800'
+                    : 'active:bg-gray-200 dark:hover:text-white',
+                  'ml-2 h-5 w-5',
+                )}
+                aria-hidden="true"
+              />
+            </Popover.Button>
+            {/* @ts-expect-error */}
```
Comment (MaggieAppleton) [inline src/components/app/header/index.tsx:440]: Can we remove this comment? (Is it needed?)

**Review by MaggieAppleton — CHANGES_REQUESTED** [review]
Pointed out some minor edits and spots to clean up, but overall this looks great! Nearly good to go

File: src/components/app/header/index.tsx (hunk 1 above)
Comment (NicollGuarnizo) [reply src/components/app/header/index.tsx]: My bad, I have fixed it

File: src/components/app/header/index.tsx (hunk 2 above)
Comment (NicollGuarnizo) [reply src/components/app/header/index.tsx]: Change done! I have renamed it as `contentSectionLinks`

File: src/components/app/header/index.tsx (hunk 3 above)
Comment (NicollGuarnizo) [reply src/components/app/header/index.tsx:440]: Yes, it's needed. Typescript throws a warning that the flyout menu it's too complex too represent, with that comment we are ignoring the warning.

**Review by MaggieAppleton — APPROVED** [review]
All looks good!

**Pull request merged**